geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureValidator.scala

```diff
@@ -0,0 +1,173 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.convert
+
+import java.util.Date
+
+import org.locationtech.geomesa.curve.BinnedTime
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+trait SimpleFeatureValidator {
+  def name: String
+  def init(sft: SimpleFeatureType): Unit
+  def validate(sf: SimpleFeature): Boolean
+  def lastError: String
```

Comment:
from a functional perspective seems like the validate method should return the error directly - although maybe that would be worse performance wise. possibly we could throw an exception - try/catch should be costless in the happy path. alternatively return an Option or Try, checking against None would probably be pretty cheap...